code/game/turfs/simulated/floor/plating/asteroid.dm

```diff
@@ -273,12 +273,14 @@ GLOBAL_LIST_INIT(megafauna_spawn_list, list(/mob/living/simple_animal/hostile/me
 			else //this is not danger, don't spawn a boss, spawn something else
 				randumb = pickweight(mob_spawn_list)
 
-		for(var/mob/living/simple_animal/hostile/H in urange(12,T)) //prevents mob clumps
-			if((ispath(randumb, /mob/living/simple_animal/hostile/megafauna) || ismegafauna(H)) && get_dist(src, H) <= 7)
+		for(var/atom/A in urange(12, T)) //prevents mob clumps
+			if(!ishostile(A) && !istype(A, /obj/structure/spawner))
+				continue
+			if((ispath(randumb, /mob/living/simple_animal/hostile/megafauna) || ismegafauna(A)) && get_dist(src, A) <= 7)
```

Comment:
remove this ispath. The ismegafauna checks for this already (and does it using istype)

Reply:
they're checking different things and ispath does not behave the same as istype